Src/Engine/Common/Containers/UniquePtr.hpp

```diff
@@ -150,12 +149,29 @@ class UniquePtr<T[], Deleter>
 };
 
 
+namespace detail {
+    template<typename>
+    constexpr bool is_unbounded_array_v = false;
+    template<typename T>
+    constexpr bool is_unbounded_array_v<T[]> = true;
+
+    template<typename>
+    constexpr bool is_bounded_array_v = false;
+    template<typename T, size_t N>
+    constexpr bool is_bounded_array_v<T[N]> = true;
+} // namespace detail
+
```

Comment:
Note that this will be removed when we transition to cpp20 (then we can just use `std::is_unbounded_array_v`)